draft-rescorla-tls-subcerts.md

```diff
@@ -300,37 +342,36 @@ TLS stack, which has the advantage of avoiding changes to security-critical and
 often delicate PKI code (though of course moves that complexity to the TLS
 stack).
 
-[[ OPEN ISSUE: If we use this syntax, how would you represent non-signature
-keys? ]]
-
-As in the above case, there would be a need for a special marker in the
-master certificate that declares that the key pair can be used to issue
-sub-certificates.
+By only allowing this signature to use TLS 1.3 signature schemes, we prevent cross
+protocol attacks such as those noted in [Jager et al.]. We could also avoid
+complicating the signature scheme selection logic by requiring the publicKey of
+the delegated credential to be the same size and format as the key in the leaf
+certificate.
```

Comment:
Right, it can do arbitrary RSA operations. My text here is wrong.

I think that requiring CAs to do anything is a non-starter. If are seriously worried about Jager-type issues in TLS 1.3, requiring a new TLS 1.3-only bit in certificates would be the way to go. This would be an adoption nightmare. The cross-protocol attack is much worse in this context than it is in general TLS 1.3, it allows an attacker to create arbitrary delegated credentials for any time period, so the argument for this to be enforced at the PKI level is strong. On the other hand the "nobody will use this" argument I think is also very strong. This will be useless for almost all existing certificates and convincing CAs to sign special certs for it is a stretch. I'm torn.

As an aside, you can make a Jager-safe certificate by simply removing the "dataEncipherment" key usage, essentially making it “sign-only”.